### PR TITLE
Add logback appender javaagent instrumentation

### PR DESCRIPTION
**Pull request by trask**
Renames `logback-1.0` instrumentation module to `logback-mdc-1.0`, and introduces `logback-appender-1.0` module.

There's no autoconfiguration of log exporter (yet), so this instrumentation doesn't benefit end users unless they explicitly configure a logging exporter via javaagent extension (this is how the tests work).

### CODE REVIEW

**Review by anuraaga — COMMENTED** [review]
Can we go ahead and add library instrumentation for logback and log4j? I guess we expect jul to be redirected to one in most cases and it's not as important there.

We already seem to be getting asked for non-javaagent logs instrumentation

https://github.com/open-telemetry/opentelemetry-java/pull/3994#issuecomment-997554707

**Comment by trask** [comment]
> Can we go ahead and add library instrumentation for logback and log4j?

We already have library instrumentation for log4j. @jack-berg mentioned last week he was working on (or thinking about writing) a logback appender.

**Comment by trask** [comment]
created #4947 to track adding a logback appender instrumentation module

File: instrumentation/logback/logback-appender-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/appender/v1_0/LogbackHelper.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.appender.GlobalLogEmitterProvider;
+import io.opentelemetry.instrumentation.api.appender.LogBuilder;
+import io.opentelemetry.instrumentation.api.appender.Severity;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.concurrent.TimeUnit;
+
+public class LogbackHelper {
```
Comment (anuraaga) [inline instrumentation/logback/logback-appender-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/appender/v1_0/LogbackHelper.java:22]:
```suggestion
public final class LogbackHelper {
```

File: instrumentation/logback/logback-appender-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/appender/v1_0/LogbackHelper.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    // time
+    long timestamp = loggingEvent.getTimeStamp();
+    builder.setEpoch(timestamp, TimeUnit.MILLISECONDS);
+
+    // level
+    Level level = loggingEvent.getLevel();
+    if (level != null) {
+      builder.setSeverity(levelToSeverity(level));
+      builder.setSeverityText(loggingEvent.getLevel().levelStr);
```
Comment (anuraaga) [inline instrumentation/logback/logback-appender-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/appender/v1_0/LogbackHelper.java:58]:
```suggestion
      builder.setSeverityText(level.levelStr);
```

**Review by anuraaga — APPROVED** [review]

File: instrumentation/logback/logback-appender-1.0/javaagent/src/test/groovy/LogbackTest.groovy [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import static io.opentelemetry.instrumentation.test.utils.TraceUtils.runUnderTrace
+import static org.assertj.core.api.Assertions.assertThat
+import static org.awaitility.Awaitility.await
+
+class LogbackTest extends AgentInstrumentationSpecification {
+
+  private static final Logger abcLogger = LoggerFactory.getLogger("abc");
+  private static final Logger defLogger = LoggerFactory.getLogger("def");
+
+  def "test logger=#loggerName method=#testMethod with exception=#exception and parent=#parent"() {
```
Comment (mateuszrzeszutek) [inline instrumentation/logback/logback-appender-1.0/javaagent/src/test/groovy/LogbackTest.groovy:21]:
```suggestion
  @Unroll
  def "test logger=#loggerName method=#testMethod with exception=#exception and parent=#parent"() {
```

File: instrumentation/logback/logback-appender-1.0/javaagent/src/test/groovy/LogbackTest.groovy [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      ],
+      [true, false],
+      [true, false]
+    ].combinations()
+
+    logger = args[0]
+    loggerName = args[1]
+    testMethod = args[2]
+    severity = args[3]
+    severityText = args[4]
```
Comment (mateuszrzeszutek) [inline instrumentation/logback/logback-appender-1.0/javaagent/src/test/groovy/LogbackTest.groovy:94]: Would 
```suggestion
    [logger, loggerName, testMethod, severity, severityText] = args
```
work?

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/logback/logback-appender-1.0/javaagent/src/test/groovy/LogbackTest.groovy (hunk 4 above)
Comment (trask) [reply instrumentation/logback/logback-appender-1.0/javaagent/src/test/groovy/LogbackTest.groovy:96]: almost! it looks like `def (logger, loggerName, testMethod, severity, severityText) = args` should work, but not in where blocks:

```
where-blocks may only contain parameterizations (e.g. 'salary << [1000, 5000, 9000]; salaryk = salary / 1000')
```

**Pull request merged**